lib/galaxy/tools/toolbox/base.py

```diff
@@ -455,8 +455,8 @@ def get_tool(self, tool_id, tool_version=None, get_all_versions=False, exact=Fal
                         for tool in rval:
                             if tool.version == tool_version:
                                 return tool
-                    # No tool matches by version, simply return the first available tool found
-                    return rval[0]
+                    # No tool matches by version, simply return the newest matching tool
+                    return sorted(rval, key=lambda x: x.version, reverse=True)[0]
```

Comment:
I think `rval[-1]` should be a faster equivalent, since `tool_lineage.get_versions()` returns ToolLineageVersion objects from oldest to newest.

Reply:
I had forgotten that!